core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -326,10 +326,14 @@ private[chisel3] object ir {
     choices:    Seq[(String, BaseModule)])
       extends Definition
   case class DefObject(sourceInfo: SourceInfo, id: HasId, className: String) extends Definition
-  case class WhenBegin(sourceInfo: SourceInfo, pred: Arg) extends Command
-  case class WhenEnd(sourceInfo: SourceInfo, firrtlDepth: Int, hasAlt: Boolean = false) extends Command
-  case class AltBegin(sourceInfo: SourceInfo) extends Command
-  case class OtherwiseEnd(sourceInfo: SourceInfo, firrtlDepth: Int) extends Command
+
+  case class When(
+    sourceInfo: SourceInfo,
+    pred:       Arg,
+    ifRegion:   VectorBuilder[Command] = new VectorBuilder[Command],
+    elseRegion: VectorBuilder[Command] = new VectorBuilder[Command])
```

Comment:
It might be worth making the `elseRegion` default to `null` and doing the `Option` boxing to make the normal API safer. Quick maths shows that an empty `VectorBuilder` costs 72-96 bytes (depending on heap size and thus if compressed pointers are in use).

Reply:
Good idea. I changed this, roughly in this direction, with https://github.com/chipsalliance/chisel/pull/4184/commits/6e53f83e60ad58998c78c84d34dd8bf6dbb52659. This kind of thing really means `case class` isn't appropriate. One nice thing about this approach is that the `unapply` can then do the conversion to `Seq[Command]` and avoid needing to do calls to `.result()` all over the place.